src/Shared/Value.elm

```diff
@@ -0,0 +1,56 @@
+module Shared.Value exposing (Value, addError, fromValue, getMaybeError, isInvalid, isValid, toValue)
+
+
+type Value a
+    = Valid a
+    | Invalid String a
+
+
+toValue : a -> Value a
+toValue =
+    Valid
```

Comment:
You expect the value to be valid straight away during the conversion to `Value`. What if we'd like to validate this value based on some validators? Would you do some changes to the `Value` module in such case or would you keep it as it is?